### PR TITLE
chore: add trace exporter db latency

### PR DESCRIPTION
**Pull request by makeavish**
closes: https://github.com/SigNoz/signoz-otel-collector/issues/77

### CODE REVIEW

File: exporter/clickhousetracesexporter/writer.go [hunk 1]
```diff
@@ -199,7 +200,20 @@ func (w *SpanWriter) writeIndexBatch(batchSpans []*Span) error {
...
 
-	return statement.Send()
+	start := time.Now()
+
+	err = statement.Send()
+	if err != nil {
+		return err
+	}
+
+	ctx, _ = tag.New(ctx,
```
Comment (nityanandagohain) [inline exporter/clickhousetracesexporter/writer.go:210]: As suggested by Srikanth, we should capture the metric even if an error occurred.

File: exporter/clickhousetracesexporter/writer.go [hunk 2]
```diff
@@ -234,7 +248,20 @@ func (w *SpanWriter) writeErrorBatch(batchSpans []*Span) error {
...
```
Comment (nityanandagohain) [inline exporter/clickhousetracesexporter/writer.go:258]: capture before catching error

File: exporter/clickhousetracesexporter/writer.go [hunk 3]
```diff
@@ -271,12 +298,19 @@ func (w *SpanWriter) writeModelBatch(batchSpans []*Span) error {
...
 	}
+	start := time.Now()
 
 	err = statement.Send()
 	if err != nil {
 		return err
 	}
 
+	ctx, _ = tag.New(ctx,
+		tag.Upsert(exporterKey, string(component.DataTypeMetrics)),
```
Comment (nityanandagohain) [inline exporter/clickhousetracesexporter/writer.go:309]: capture before catching error

File: exporter/clickhousetracesexporter/writer.go (hunk 1 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/writer.go:210]: ok

**Review by nityanandagohain — APPROVED** [review]

File: exporter/clickhousetracesexporter/writer.go [hunk 4]
```diff
@@ -199,7 +200,19 @@ func (w *SpanWriter) writeIndexBatch(batchSpans []*Span) error {
 		}
 	}
 
-	return statement.Send()
+	start := time.Now()
+
+	err = statement.Send()
+
+	ctx, _ = tag.New(ctx,
+		tag.Upsert(exporterKey, string(component.DataTypeMetrics)),
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/writer.go:208]: This should be traces
```suggestion
		tag.Upsert(exporterKey, string(component.DataTypeTraces)),
```

File: exporter/clickhousetracesexporter/writer.go [hunk 5]
```diff
@@ -199,7 +200,19 @@ func (w *SpanWriter) writeIndexBatch(batchSpans []*Span) error {
...
+		tag.Upsert(tableKey, w.indexTable),
+	)
+	stats.Record(ctx, writeLatencyMillis.M(int64(time.Since(start).Milliseconds())))
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/writer.go:215]:
```suggestion
	return err
```

File: exporter/clickhousetracesexporter/writer.go [hunk 6]
```diff
@@ -234,7 +247,19 @@ func (w *SpanWriter) writeErrorBatch(batchSpans []*Span) error {
 		}
 	}
 
-	return statement.Send()
+	start := time.Now()
+
+	err = statement.Send()
+
+	ctx, _ = tag.New(ctx,
+		tag.Upsert(exporterKey, string(component.DataTypeMetrics)),
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/writer.go:255]: Same here

File: exporter/clickhousetracesexporter/writer.go [hunk 7]
```diff
@@ -271,12 +296,17 @@ func (w *SpanWriter) writeModelBatch(batchSpans []*Span) error {
 
 		usage.AddMetric(metrics, *span.Tenant, 1, int64(len(serialized)))
 	}
+	start := time.Now()
 
 	err = statement.Send()
+	ctx, _ = tag.New(ctx,
+		tag.Upsert(exporterKey, string(component.DataTypeMetrics)),
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/writer.go:303]: Same here

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**